faunadb/errors.py

```diff
@@ -103,18 +100,24 @@ class ErrorData(object):
   @staticmethod
   def from_dict(dct):
     code = get_or_invalid(dct, "code")
-    return ValidationFailed(dct) if code == "validation failed" else ErrorData(dct, code)
+    description = get_or_invalid(dct, "description")
+    position = dct.get("position")
+    if code == "validation failed":
+      failures = [Failure.from_dict(failure) for failure in get_or_invalid(dct, "failures")]
+      return ValidationFailed(description, position, failures)
+    else:
+      return ErrorData(code, description, position)
 
-  def __init__(self, dct, code):
+  def __init__(self, code, description, position):
     self.code = code
     """Error code. See the many error codes `here <https://faunadb.com/documentation#errors>`__."""
```

Comment:
I know it's not part of this pr, but "the many error codes"? Maybe just "all the error codes".